app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+#
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:success] = 'Signed Up'
+      redirect_to @user
+    else
+      render 'static_pages/error'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+    redirect_to root_path && return unless @user == current_user
+    # how do I insert an error message?
```

Comment:
Mainly so I don't forget. My specific issue here is that I am not sure where to place the flash[:alert]. It doesn't seem to belong after the redirect and when I attempted to add it in after the @user == current_user check, I get an error. All in all the code seems clumsy but I am not sure how to improve it.